File: src/dataset/typescript/allergies_exercism-wackerow/claude/ClaudeSonnet4_allergies_v2.ts
export default class Allergies {
  private _allergies: number;
  private static readonly _key: readonly string[] = [
    'eggs',
    'peanuts',
    'shellfish',
    'strawberries',
    'tomatoes',
    'chocolate',
    'pollen',
    'cats',
  ];
  private static readonly _keyMap = new Map(
    Allergies._key.map((allergy, index) => [allergy, 1 << index])
  );

  constructor(allergies: number) {
    this._allergies = allergies & 255; // 2^8 - 1 = 255
  }

  public allergicTo(allergy: string): boolean {
    const mask = Allergies._keyMap.get(allergy);
    return mask !== undefined && (this._allergies & mask) !== 0;
  }

  public list(): string[] {
    const result: string[] = [];
    for (let i = 0; i < Allergies._key.length; i++) {
      if (this._allergies & (1 << i)) {
        result.push(Allergies._key[i]);
      }
    }
    return result;
  }
}

File: src/dataset/python/allergies_exercism-ThomasZumsteg/claude/ClaudeSonnet4_allergies_v2.py
"""Tracks allergies"""

class Allergies(object):
    """Generates and stores allergies from a given score"""

                   # Allergie       # score
    allergie_list = ['eggs',        # 1
                     'peanuts',     # 2
                     'shellfish',   # 4
                     'strawberries',# 8
                     'tomatoes',    # 16
                     'chocolate',   # 32
                     'pollen',      # 64
                     'cats',        # 128
                    ]

    def __init__(self, score):
        """Generates a list of allergies from a score"""
        self.score = score & 255  # Mask to only consider valid bits
        self.lst = None

    def _get_allergies(self):
        """Lazy initialization of allergies list"""
        if self.lst is None:
            self.lst = [allergie
                        for i, allergie in enumerate(Allergies.allergie_list)
                        if self.score & (1 << i)]
        return self.lst

    def is_allergic_to(self, allergie):
        """Tests if allergie is on the list"""
        try:
            index = Allergies.allergie_list.index(allergie)
            return bool(self.score & (1 << index))
        except ValueError:
            return False

    @property
    def lst(self):
        return self._lst

    @lst.setter
    def lst(self, value):
        self._lst = value

    @lst.getter
    def lst(self):
        return self._get_allergies()